offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -227,37 +233,39 @@ class OfferToPDFConverter implements OfferExporter {
     void setSelectedItems() {
         // Let's clear the existing item template content first
         htmlContent.getElementById("product-items-1").empty()
-        //and remove the footer on the first page
-        //htmlContent.getElementById("grid-table-footer").remove()
+        htmlContent.getElementById("product-items-2").empty()
+        // To also remove the styling of the div elements they have to be removed
+        htmlContent.getElementById("grid-table-footer").remove()
+        htmlContent.getElementById("template-page-break").remove()
 
         List<ProductItem> productItems = offer.items
 
         //Initialize Number of table
         tableCount = 1
 
-        //Initialize Count of ProductItems in table
+        //Initialize Count which will be used to determine how many Items are stored in the table
         tableItemsCount = 1
-        int maxTableItems = 8
+        //The maximum Items is currently set manually and estimated by eye,
+        //since there is no way to properly account for differences of lengths of the productItem description
+        int maxTableItems = 13
 
         //Group ProductItems into Data Generation Data Analysis and Data & Project Management Categories
         Map productItemsMap = groupItems(productItems)
 
         //Generate Product Table for each Category
         generateProductTable(productItemsMap, maxTableItems)
-
         //Append total cost footer
+        String elementId = "product-items" + "-" + tableCount
         if (tableItemsCount > maxTableItems) {
             //If currentTable is filled with Items generate new one and add total pricing there
             ++tableCount
-            String elementId = "product-items" + "-" + tableCount
-            htmlContent.getElementById("item-table-grid").append(ItemPrintout.tableHeader(elementId))
-            htmlContent.getElementById("item-table-grid")
-                    .append(ItemPrintout.tableFooter(offer.overheadRatio, offer.getSelectedCustomerAffiliation()))
-        } else {
-            //otherwise add total pricing to table
-            htmlContent.getElementById("item-table-grid")
-                    .append(ItemPrintout.tableFooter(offer.overheadRatio, offer.getSelectedCustomerAffiliation()))
+            htmlContent.getElementById(elementId).append(ItemPrintout.pageBreak())
+            elementId = "product-items" + "-" + tableCount
+            htmlContent.getElementById("item-table-grid").append(ItemPrintout.createNewTable(elementId))
+            htmlContent.getElementById(elementId).append(ItemPrintout.tableHeader())
```

Comment:
I think there should be a method that starts a new page with a fresh table just like what you do here. reading the method name makes it more obvious what happens here

```suggestion
           generateNewPageWithTotalItemPricing(String elementId)
```

or something like that

Reply:
How can you decide about starting a new page based on the tableItemsCount? This is the number for all items that are added to the html yet, right? The MaxTableItems says how many items can be stored on the page. If you have more than 2 pages it would not work, right?

Reply:
The `tableItemsCount` is reset to one after the `MaxTableItems `is reached and the new items are then added to a new table which is appended to the "item-table-grid". I agree that the naming Schema is confusing though❓.
I like the idea of outsourcing the edge cases into different methods and will do so on monday 😄